packages/headless/src/controllers/product-listing/headless-product-listing.ts

```diff
@@ -151,9 +241,66 @@ export function buildProductListing(
   };
 }
 
+export function buildProductListingV2(
```

Comment:
I guess this controller could actually be called `productList` instead of `listing ? 

In the same vein that we have a `resultList`, it's not that different.

And the main function that are linked to this controller are potentially... a bit illogical (`refresh`).

What we could do is picture the whole thing like this:

- One productListing engine
- One or multiple facets bound to that engine
- One productList bound to that engine
- One pager bound to that engine
- Sort bound to that engine
- etc...

The product listing engine would expose the `refresh()`, `setProductListingId()`, `setURL()`, etc.. functions since those are global operation (ie: it affects every controller bound to that engine).

ProductList could expose function related to "infinite scrolling" (`fetchMoreProducts`), since it's the functionality you would expect on a list of product.

Reply:
We should do a quick sync about that.

I'll start a design doc and share it with you and @fbeaudoincoveo for now.

Reply:
I have the very minimal base for a commerce sub-package here fwiw:
https://github.com/coveo/ui-kit/compare/master...feat-headless-commerce